src/services/viewFormatter.services.ts

```diff
@@ -0,0 +1,129 @@
+/****************** PARSING STRATEGY INTERFACE **********************/
+
+import {
+  countHighSeverityCves,
+  readPackagesDashboard,
+  readVulnerabilitiesDashboard,
+  countVulnerabilities,
+  countPackages,
+  countHighRiskCves,
+} from '../utils/storageFacade.utils';
+import {
+  Vulnerability,
+  DisplayPackage,
+  PackageViewParam,
+  VulnerabilityViewParam,
+  severityRating,
+} from '../utils/types.utils';
+
+enum ViewType {
+  PACKAGE = 'component',
+  VULNERABILITY = 'vulnerability',
+}
+
+interface DashboardView {
+  type: ViewType;
+  stats: {
+    Components_Detected: number;
+    Vulnerabilities_Identified: number;
+    High_Severity_Vulnerabilities: number;
+    High_Risk_Vulnerabilities: number;
+  };
+  data: Vulnerability[] | DisplayPackage[];
+}
+
+export interface DashboardRequest {
+  viewType: ViewType;
+  sessionId: number;
+  riskFilters: string[];
+  severityFilters: string[];
+  page: number;
+  sortParam: string;
+}
+
+export async function getView(req: DashboardRequest) {
+  let viewFormatter: JsonFormatter;
+  if (req.viewType == ViewType.PACKAGE) {
+    viewFormatter = packageViewFormatter;
+  } else {
+    viewFormatter = vulnerabilityViewForamtter;
+  }
+  return viewFormatter(req);
+}
+
+/**
+ *
+ *
+ * @interface JsonFormatter
+ */
+interface JsonFormatter {
+  (req: DashboardRequest): Promise<DashboardView>;
+}
+
+function decodePackageViewParam(s: string): PackageViewParam {
```

Comment:
add a descriptive line for these two decode functions - where is decoding required?